src/fides/api/ctl/migrations/versions/63b482f5b49b_upgrade_to_fideslang_1_4.py

```diff
@@ -0,0 +1,93 @@
+from fides.api.oauth.roles import OWNER
+from fides.core.config import CONFIG
+
+# revision identifiers, used by Alembic.
+revision = "63b482f5b49b"
+down_revision = "2661f31daffb"
+branch_labels = None
+depends_on = None
+
+# The `key` is the old value, the `value` is the new value
+data_use_updates: Dict[str, str] = {
+    "third_party_sharing.payment_processing": "essential.service.payment_processing"
+    "third_party_sharing.fraud_detection",
+    "essential.service.fraud_detection"
+    "advertising.first_party": "marketing.advertising.first_party",
+    "advertising.first_party.contextual": "marketing.advertising.first_party.contextual",
+    "advertising.first_party.personalized": "marketing.advertising.first_party.targeted",
+    "advertising.third_party": "marketing.advertising.third_party.targeted",
+    "advertising.third_party.personalized": "marketing.advertising.third_party.targeted",
+    "third_party_sharing.personalized_advertising": "marketing.advertising.third_party.targeted",
+    "provide": "essential",
+    "provide.service": "essential.service",
+}
+
+
+async def update_privacy_declarations() -> None:
+    async with async_session() as session:
+        resources = await list_resource(PrivacyDeclaration, session)
+
+        for resource in resources:
+            current_data_use = resource.data_use
+
+            if current_data_use in data_use_updates.keys():
+                resource.data_use = data_use_updates[resource.data_use]
+
+        await upsert_resources(PrivacyDeclaration, resources, session)
+
+        # Since we can't write tests for data migrations, we do it live
+        existing_resources = await list_resource(PrivacyDeclaration, session)
+        assert not all(
+            [
+                False if resource.data_use in data_use_updates.keys() else True
+                for resource in existing_resources
+            ]
+        ), "ERROR: Data Use Migration for Fideslang v1.4 failed for Privacy Declarations!"
+
+
+async def update_policy_rules() -> None:
+    async with async_session() as session:
+        resources = await list_resource(PolicyCtl, session)
+        for resource in resources:
+            for rule in resource.rules:
+                current_data_use = rule.data_use
+                if current_data_use in data_use_updates.keys():
+                    rule.data_use = data_use_updates[resource.data_use]
+        await upsert_resources(PolicyCtl, resources, session)
+
+        # Since we can't write tests for data migrations, we do it live
+        existing_resources = await list_resource(PolicyCtl, session)
+        assert not all(
+            [
+                False if rule.data_use in data_use_updates.keys() else True
+                for resource in existing_resources
+                for rule in resource
+            ]
+        ), "ERROR: Data Use Migration for Fideslang v1.4 failed for Policy Rules!"
+
+
+def upgrade():
```

Comment:
@adamsachs @SteveDMurphy looking for an early review here to double-check my thinking for this data use migration

Does this method make sense? Can you think of any other pattern that might work better?

I haven't finalized the change over in `fideslang` but I was planning on using this design to do the in-place database upgrade. Luckily data uses aren't nearly as widely used as categories so there aren't many places that need updating here

Reply:
i like this! it's pretty easy and clean to follow. would you not want to do a reverse mapping for the downgrade?

also, i had a question on the `assert`, i feel like it may be slightly off? but i may be reading it wrong. generally i do like the extra safety in asserting what we expect here, that's a nice touch.

Reply:
Definitely going to add the reverse downgrade, but didn't want to get too deep since I'd use the same paradigm there!

Will add now and clean up the bug you found, thanks for the early and quick review!